src/Token/LcobucciFactory.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Mercure Component project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Mercure\Token;
+
+use Lcobucci\JWT\Configuration;
+use Lcobucci\JWT\Signer;
+use Lcobucci\JWT\Signer\Key;
+use Symfony\Component\Mercure\Exception\InvalidArgumentException;
+
+if (!class_exists(Key\InMemory::class)) {
+    throw new \LogicException('You cannot use "Symfony\Component\Mercure\Token\LcobucciFactory" as the "lcobucci/jwt" package is not installed. Try running "composer require lcobucci/jwt".');
+}
+
+final class LcobucciFactory implements TokenFactoryInterface
+{
+    /**
+     * @var array<string, class-string<Signer>>
+     */
+    public const SIGN_ALGORITHMS = [
+        'hmac.sha256' => Signer\Hmac\Sha256::class,
+        'hmac.sha384' => Signer\Hmac\Sha384::class,
+        'hmac.sha512' => Signer\Hmac\Sha512::class,
+        'ecdsa.sha256' => Signer\Ecdsa\Sha256::class,
+        'ecdsa.sha384' => Signer\Ecdsa\Sha384::class,
+        'ecdsa.sha512' => Signer\Ecdsa\Sha512::class,
+        'rsa.sha256' => Signer\Rsa\Sha256::class,
+        'rsa.sha384' => Signer\Rsa\Sha384::class,
+        'rsa.sha512' => Signer\Rsa\Sha512::class,
+    ];
+
+    private $configurations;
+
+    public function __construct(string $secret, string $algorithm = 'hmac.sha256')
+    {
+        if (!\array_key_exists($algorithm, self::SIGN_ALGORITHMS)) {
```

Comment:
Can't we do something dynamic such as:

```suggestion
        list($type, $alg) = explode('.', $algorithm, 2);
        if (!isset($alg) || !class_exists($signerClass = sprintf('Lcobucci\JWT\Signer\%s\%s', ucfirst($type), ucfirst($alg))) {
```

This will prevent to have to update the component every time an algorithm is added upstream (and this will simplify the code).

Reply:
this will trigger a warning for something like `"foo"` ( https://3v4l.org/UN6uD ), also, we can't include supported algorithms in the exception message ( which is would be really helpful for people that made a typo - e.g: `esca.sha512` instead of `ecdsa.sha512` )

Reply:
```
        $a = explode('.', $algorithm, 2);
        if (!isset($a[0]) || !class_exists($signerClass = sprintf('Lcobucci\JWT\Signer\%s\%s', ucfirst($a[0]), ucfirst($a[1]))) {
```

then
```